### PR TITLE
added an enum to combine preference types user + hub - fixes issue wi…

### PR DESCRIPTION
**Pull request by techsmyth**
…th obtaining type field on a definition

### CODE REVIEW

File: src/common/enums/preference.type.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { registerEnumType } from '@nestjs/graphql';
```
Comment (valentinyanakiev) [inline src/common/enums/preference.type.ts:1]: You already have the preferences, try replacing with the following, avoiding code duplication and forcing future compatibility when enums are changed:

```suggestion
import { registerEnumType } from '@nestjs/graphql';
import { UserPreferenceType } from './user.preference.type';
import { HubPreferenceType } from './hub.preference.type';

export const PreferenceType = {
  ...UserPreferenceType,
  ...HubPreferenceType,
};

export type PreferenceType = UserPreferenceType | HubPreferenceType;

registerEnumType(PreferenceType, {
  name: 'PreferenceType',
});

```

File: src/domain/common/preference/preference.definition.interface.ts [hunk 2]
```diff
@@ -21,7 +22,7 @@ export abstract class IPreferenceDefinition extends IBaseAlkemio {
   })
   description!: string;
 
-  @Field(() => UserPreferenceType, {
+  @Field(() => PreferenceType, {
     description: 'The type of the Preference, specific to the Entity it is on.',
   })
   type!: UserPreferenceType | HubPreferenceType;
```
Comment (valentinyanakiev) [inline src/domain/common/preference/preference.definition.interface.ts]:
```suggestion
  type!: PreferenceType;
```

**Review by valentinyanakiev — CHANGES_REQUESTED** [review]
Please review my suggested changes, I think we can avoid copy/pasting and code duplication without breaking functionality. I have created a server-1792-b branch so you can check out the changes.

File: src/common/enums/preference.type.ts (hunk 1 above)
Comment (techsmyth) [reply src/common/enums/preference.type.ts:1]: Feedback is incorporated in base branch

File: src/domain/common/preference/preference.definition.interface.ts (hunk 2 above)
Comment (techsmyth) [reply src/domain/common/preference/preference.definition.interface.ts]: Feedback is incorporated in base branch

**Review by valentinyanakiev — APPROVED** [review]

**Pull request merged**